extensions/kinesis/src/test/java/com/hazelcast/jet/kinesis/KinesisIntegrationTest.java

```diff
@@ -185,6 +191,41 @@ public void customProjection() {
         }
     }
 
+    @Test
+    public void testCustomSinkExecutorService() throws Exception {
+        HELPER.createStream(1);
+
+        threadCounter.set(0);
+        SupplierEx<ExecutorService> sinkExecutorSupplier = () -> newFixedThreadPool(
+                1,
+                r -> new Thread(r, "kinesis-sink-thread-" + threadCounter.getAndIncrement())
+        );
+        Sink<Map.Entry<String, byte[]>> sink = kinesisSink()
+                .withExecutorServiceSupplier(sinkExecutorSupplier)
+                .build();
+
+        sendMessages(MESSAGES, sink);
+
+        try {
+            Pipeline pipeline = Pipeline.create();
+            pipeline.readFrom(kinesisSource().build())
+                    .withoutTimestamps()
+                    .groupingKey(key -> "sameKeyAllEntries")
+                    .rollingAggregate(counting())
+                    .apply(assertCollectedEventually(ASSERT_TRUE_EVENTUALLY_TIMEOUT, windowResults -> {
+                        assertTrue((long) windowResults.size() == MESSAGES
+                                && threadCounter.get() == MEMBER_COUNT);
```

Comment:
I think it makes more sense to move the threadCounter assert just after `sendMessages`. No need to assert it inside a job.